packages/browser-wallet/src/shared/utils/time-helpers.ts

```diff
@@ -2,6 +2,23 @@ export function secondsToDaysRoundedDown(seconds: bigint | undefined): bigint {
     return seconds ? seconds / (60n * 60n * 24n) : 0n;
 }
 
+type TimeAndUnit = { time: bigint; unit: string };
+
+function msToTime(ms: bigint | undefined): TimeAndUnit {
+    if (!ms) return { time: 0n, unit: 'minute' };
+    const minutes = ms / (1000n * 60n);
+    const hours = ms / (1000n * 60n * 60n);
+    const days = ms / (1000n * 60n * 60n * 24n);
+    if (minutes < 60n) return { time: minutes, unit: 'minute' };
+    if (hours < 24n) return { time: hours, unit: 'hour' };
+    return { time: days, unit: 'day' };
+}
```

Comment:
// I suggest having these time constants somewhere globally defined since they are likely reused already in the code base. 
    const MILLISECONDS_PER_DAY = 1000n * 60n * 60n * 24n;
    const MILLISECONDS_PER_HOUR = 1000n * 60n * 60n;
    const MILLISECONDS_PER_MINUTE = 1000n * 60n;

```suggestion
function msToTime(ms: bigint | undefined): TimeAndUnit {
    if (!ms) return { time: 0n, unit: 'minute' };
   
    if (ms < MILLISECONDS_PER_HOUR) return { time: ms / MILLISECONDS_PER_MINUTE, unit: 'minute' };
    if (ms < MILLISECONDS_PER_DAY) return { time: ms / MILLISECONDS_PER_HOUR, unit: 'hour' };
    return { time: ms / MILLISECONDS_PER_DAY, unit: 'day' };

}
```